src/ansys/dpf/post/elements.py

```diff
@@ -0,0 +1,240 @@
+"""This module contains ElementList, ElementType and Element classes."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import List, Union
+
+import ansys.dpf.core as dpf
+from ansys.dpf.core import elements as core_elements
+from ansys.dpf.core import errors
+from ansys.dpf.core.nodes import Node
+
+
+class ElementType:
+    """Wrapper type to instantiate an ElementDescriptor from an int."""
+
+    def __init__(self, arg: Union[dpf.ElementDescriptor, int]):
+        """Constructs an ElementType from an existing descriptor or enum id."""
+        self._el_desc = arg
+        if isinstance(arg, int):
+            self._el_desc = dpf.element_types.descriptor(arg)
+
+        if not isinstance(self._el_desc, dpf.ElementDescriptor):
+            raise TypeError("Given argument is not an int nor an ElementDescriptor")
+
+    @property
+    def elem_type_id(self) -> dpf.element_types:
+        """Element type in the element_types enum."""
+        return self._el_desc.enum_id
+
+    @property
+    def description(self) -> str:
+        """Specifies the element geometry and integration order."""
+        return self._el_desc.description
+
+    @property
+    def name(self) -> str:
+        """Short name of the element type."""
+        return self._el_desc.name
+
+    @property
+    def shape(self) -> str:
+        """Can be ``"solid"``,``"shell"`` or ``"beam"``."""
+        return self._el_desc.shape
+
+    @property
+    def num_corner_nodes(self) -> int:
+        """Returns the number of corner nodes."""
+        return self._el_desc.n_corner_nodes
+
+    @property
+    def num_mid_nodes(self) -> int:
+        """Returns the number of middle nodes."""
+        return self._el_desc.n_mid_nodes
+
+    @property
+    def num_nodes(self) -> int:
```

Comment:
How do these methods behave with Polys? (where we don't know the numbers)

Reply:
@rafacanton it's my next step, to adapt these to fluid meshes, in a next PR